doc/manual/syntax.md

```diff
@@ -200,6 +200,77 @@ def concat(str_array, log=false):
   return res"
 ```
 
+#### Symbolic strings
+
+Symbolic strings are another type of special strings in Nickel. For example, you
+could encounter the following code in a library for using Nickel with Nix:
+
+```nickel
+{
+  args = [
+      "-c",
+      nix-s%"
+        %{inputs.gcc}/bin/gcc %{inputs.hello} -o hello
+        %{inputs.coreutils}/bin/mkdir -p $out/bin
+        %{inputs.coreutils}/bin/cp hello $out/bin/hello
+      "%,
+  ],
+  ..
+}
+```
+
+Lines 4 to 8 define a symbolic string. Values `inputs.gcc`, `inputs.hello`, etc.
+aren't actually strings, but arbitrary records. For concrete use-cases, some
+tools require manipulating values are not yet known at the time of evaluation
+(e.g. Terraform's computed values) or to perform additional dependency tracking
+([Nix string context][nix-string-context]).
+
+The idea behind symbolic strings is to offer a string-like syntax, but without
+evaluating the expression as a string. Instead, the expression is returned in a
+symoblic form - in practice, an array of fragments, where each fragment is
+either a string or an arbitrary value that has been interpolated - and let the
+specific library (Terraform-Nickel, Nix-Nickel, etc.) process it.
+
+The prefix of a symbolic string is a sequence of lower-case alphabetic
+characters followed by `-s`. Prefixes don't have any meaning for Nickel: they're
+a tag used by librairies consuming symbolc strings to distinguish between
+several types of symbolic strings. Tags are a visual marker for the programmer
+as well.
```

Comment:
The prefix actually need only match `[a-zA-Z][_a-zA-Z0-9-']*`

Reply:
It's in fact an identifier that doesn't start with `_` and ends with `-s`, I think.

Reply:
Yes, I think that's it.